lightning/src/ln/channelmanager.rs

```diff
@@ -10367,14 +10375,25 @@ where
 				};
 
 				match result {
-					Ok(()) => ResponseInstruction::NoResponse,
-					Err(e) => match responder {
-						Some(responder) => responder.respond(OffersMessage::InvoiceError(e)),
-						None => {
-							log_trace!(self.logger, "No reply path for sending invoice error: {:?}", e);
-							ResponseInstruction::NoResponse
-						},
-					},
+					Err(err) => {
+						match responder {
+							Some(responder) => {
+								abandon_if_payment(context);
+								responder.respond(OffersMessage::InvoiceError(err))
+							}
+							None => {
+								abandon_if_payment(context);
+								log_trace!(
+									self.logger,
+									"A error response was generated, but there is no reply_path specified \
+									for sending the response. Error: {}",
+									err
+								);
+								return ResponseInstruction::NoResponse;
+							}
+						}
+					}
```

Comment:
Since this arm uses a single expression -- the `match` -- it can be moved up in order to reduce the level indentation.

```rust
Err(err) => match responder {
	// ...
},

Reply:
Done! Cleaned the code in [3285830](https://github.com/lightningdevkit/rust-lightning/commit/328583045b9aadb1e35e6d3b180d0a30ca526a63)
Thanks for the suggestion!